### PR TITLE
Check if attempts are allowed and spare time is available before answer is checked

### PR DESCRIPTION
**Pull request by ankitjavalkar**
Changed the function name because - https://www.datacamp.com/community/tutorials/role-underscore-python#SLU

### CODE REVIEW

File: yaksh/views.py [hunk 1]
```diff
@@ -844,6 +851,7 @@ def check(request, q_id, attempt_num=None, questionpaper_id=None,
         if current_question.type in ['code', 'upload']:
             if (paper.time_left() <= 0 and not
                     paper.question_paper.quiz.is_exercise):
+                print(">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>")
```
Comment (adityacp) [inline yaksh/views.py:854]: Remove Print

Comment (ankitjavalkar) [reply yaksh/views.py]: Fixed

**Comment by prathamesh920** [comment]
Implementing this will prevent checking of answer when the quiz auto submits due to time up.
If the user has selected an answer then it should be checked on submit or auto-submit.

Also, I think ```can_attempt_now``` method is not needed here, as this view is checking the answer for the already existing attempt and not requesting for a new one.

File: yaksh/views.py [hunk 2]
```diff
@@ -738,10 +738,17 @@ def check(request, q_id, attempt_num=None, questionpaper_id=None,
         question_paper=questionpaper_id,
         course_id=course_id
     )
+    questionpaper = QuestionPaper.objects.get(id=questionpaper_id)
```
Comment (prathamesh920) [inline yaksh/views.py:741]: This is not needed

Comment (prathamesh920) [inline yaksh/views.py:741]: This is not needed

**Comment by prathamesh920** [comment]
LGTM

**Comment by CruiseDevice** [comment]
LGTM. Coverage seems to be decreasing, pep8 issues and also build fails. Please fix it.

**Comment by ankitjavalkar** [comment]
Finally passes tests after 5 odd rebuilds on travis :(

**Pull request merged**